src/System.Net.WebProxy/src/System/Net/WebProxy.cs

```diff
@@ -0,0 +1,238 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Globalization;
+using System.Net.NetworkInformation;
+using System.Runtime.Serialization;
+using System.Text.RegularExpressions;
+
+namespace System.Net
+{
+    [Serializable]
+    public class WebProxy : ISerializable
+    {
+        private ArrayList _bypassList;
+        private Regex[] _regExBypassList;
+
+        public WebProxy() : this((Uri)null, false, null, null) { }
+
+        public WebProxy(Uri Address) : this(Address, false, null, null) { }
+
+        public WebProxy(Uri Address, bool BypassOnLocal) : this(Address, BypassOnLocal, null, null) { }
+
+        public WebProxy(Uri Address, bool BypassOnLocal, string[] BypassList) : this(Address, BypassOnLocal, BypassList, null) { }
+
+        public WebProxy(Uri Address, bool BypassOnLocal, string[] BypassList, ICredentials Credentials)
+        {
```

Comment:
I realize that it would fail in your example. What I am suggesting is a breaking change.  But perhaps no one will really be negatively affected by it. And in the long run, it is the right change.